cmd/state-svc/internal/server/server.go

```diff
@@ -61,6 +65,12 @@ func (s *Server) Start() error {
 	return s.httpServer.Start(s.listener.Addr().String())
 }
 
+func (s *Server) quit() {
+	s.done <- true
+	close(s.done)
```

Comment:
This doesn't look quite right.  As the done channel is non-blocking, it wouldn't wait for the reader to consume the true event, and just immediately close the channel.  That of course, indicates the quite event, by waking up the channel, but most likely the client will not retrieve the `true`.

I think that is okay though.  And we should just remove line 69 (and the buffer size of 1 on the channel).